### DISCUSSION TITLE
Do not use a custom config-key for ApplicationInsights' instrumentation-key

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.templates#536] [General]
When creating a new Web API using the Arcus template, some code is generated to setup Application Insights logging.

The template code expects the instrumentation-key to be in a custom configuration-setting (`Telemetry:ApplicationInsights:InstrumentationKey`) as you can see [here](https://github.com/arcus-azure/arcus.templates/blob/ada3dfe76812107a9a584b18f6569f7eb9dc3054/src/Arcus.Templates.WebApi/Program.cs#L55).

However, I think it is better if we use one of the 'default' configuration values.   There are 2 settings documented in Microsoft's docs ( `ApplicationInsights:InstrumentationKey` and `APPINSIGHTS_INSTRUMENTATIONKEY`.

I've found that, when using `APPINSIGHTS_INSTRUMENTATIONKEY`, the Azure Portal also recognizes that App Insights logging has been setup.   I'd be in favor of using the `APPINSIGHTS_INSTRUMENTATIONKEY` setting in Arcus templates as wel, instead of that custom setting.

Any thoughts @gverstraete  @scleynen  ?

**Comment by gverstraete** [comment]
I can agree with that, but isn't there a connectionstring alternative?

Ref: https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=net

**Reply by fgheysels** [reply]
I have another issue for that , as I think these are 2 separate items.
I think the quick win here , is to change our Arcus templates to use the more 'standard' config-setting.
In the meantime, we should add an additional extension method in Arcus Observability to support AppInsights' connectionstrings

https://github.com/arcus-azure/arcus.observability/issues/354

**Comment by scleynen** [comment]
Yes indeed, in the projects where I used the Arcus Library I actually added
the same key twice. Once with the custom config and once with the microsoft
one to make the portal work indeed.

So I think it is a good idea to standardize on it

**Comment by fgheysels** [comment]
I think this issue has already been handled / fixed ?
